### ISSUE TITLE
Add a TextInput component

### ISSUE THREAD

**Issue by kmeht** [bmun/huxley#474]
Currently, in the registration view, we represent all of our text inputs as the raw HTML elements:
```
<input
  type="text"
  placeholder="Username"
  valueLink={this.linkState('username')}
/>
{and then we render the errors below it...}
```
Instead, let's add a `TextInput` React component, so if we ever decide to change the appearance of the text inputs, we wouldn't have to go through and modify them all by hand. In the registration component, we'd write:
```
<TextInput
  errors={this.state.errors['username']}
  label="Username"
  valueLink={this.linkState('username')}
/>
```

**Comment by kmeht** [comment]
#289 is an example of a PR that extracts a component out of RegistrationView.

**Comment by nathanielparke** [comment]
Taking care of this because it relates to React v15.0.0 upgrade

**Comment by nathanielparke** [comment]
addressed in #490

**Comment by kmeht** [comment]
Fixed in #490.
